src/Xamarin.Android.Tools.AndroidSdk/Sdks/AndroidSdkBase.cs

```diff
@@ -139,7 +160,63 @@ protected virtual IEnumerable<string> GetAllAvailableAndroidNdks ()
 			foreach (var sdk in GetAllAvailableAndroidSdks ()) {
 				if (sdk == AndroidSdkPath)
 					continue;
-				yield return Path.Combine (sdk, "ndk-bundle");
+				yield return FindBestNDK (sdk);
+			}
+		}
+
+		string FindBestNDK (string androidSdkPath)
```

Comment:
Nit-pick, and I'm not sure I should be saying this, *but*…

I'd prefer if this were "more pieces" with "clearer ordering":

```csharp
IEnumerable<string> GetSdkNdkPaths (string androidSdkPath)
{
    var ndk = Path.Combine (androidSdkPath, "ndk");
    if (Directory.Exists (ndk)) {
        foreach (string ndkPath in Directory.EnumerateDirectories (ndk, SearchOption.TopDirectoryOnly)) {
            yield return ndkDir;
        }
    }
    ndk = Path.Combine (androidSdkPath, "ndk-bundle");
    if (Directory.Exists (ndk))
        yield return ndk;
}

IEnumerable<string> OrderNdks (IEnumerable<string> ndkDirs)
{
    return from ndk in ndkDirs
        let ndkVersion = LoadNDKVersion (ndk)
        where ndkVersion > MaximumCompatibleNDKMajorVersion && ndkVersion < MinimumCompatibleNDKMajorVersion
        orderby ndkVersion descending
        select ndk;
}

string GetBestSdkNdkPath (string androidSdkPath)
{
    return OrderNdks (GetSdkNdkPaths (androidSdkPath)).FirstOrDefault ();
}
```

Reply:
Let's not do this right now. This entire code needs to be revised and refactored - most of the stuff can be done in the base class, for both Unix and Windows (and should be). Let's postpone it for some near future